garage/tf/policies/gaussian_mlp_policy_with_model.py

```diff
@@ -0,0 +1,121 @@
+     threshold value, to avoid numerical issues
+    :param std_hidden_nonlinearity:
+    :param hidden_nonlinearity: nonlinearity used for each hidden layer
+    :param output_nonlinearity: nonlinearity for the output layer
+    :param mean_network: custom network for the output mean
+    :param std_network: custom network for the output log std
+    :param std_parametrization: how the std should be parametrized. There
+     are a few options:
+        - exp: the logarithm of the std will be stored, and applied a
+         exponential transformation
+        - softplus: the std will be computed as log(1+exp(x))
+    :return:
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 name='GaussianMLPPolicy',
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=tf.nn.tanh,
+                 output_nonlinearity=None,
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_hidden_nonlinearity=tf.nn.tanh,
+                 std_output_nonlinearity=None,
+                 std_parameterization='exp'):
+        assert isinstance(env_spec.action_space, Box)
+
+        self.name = name
+        self._env_spec = env_spec
+        obs_dim = env_spec.observation_space.flat_dim
+        action_dim = env_spec.action_space.flat_dim
+
+        state_input = tf.placeholder(tf.float32, shape=(None, obs_dim))
+        self.model = GaussianMLPModel(
+            output_dim=action_dim,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            output_nonlinearity=output_nonlinearity,
+            learn_std=learn_std,
+            adaptive_std=adaptive_std,
+            std_share_network=std_share_network,
+            init_std=init_std,
+            min_std=min_std,
+            max_std=max_std,
+            std_hidden_sizes=std_hidden_sizes,
+            std_hidden_nonlinearity=std_hidden_nonlinearity,
+            std_output_nonlinearity=std_output_nonlinearity,
+            std_parameterization=std_parameterization)
+
+        with tf.variable_scope(name, reuse=False):
+            _, self._mean_var, self._log_std_var, _, self._dist = (
+                self.model.build(state_input))
+
+            self._f_dist = lambda x: tf.get_default_session().run(
+                [self._mean_var, self._log_std_var],
+                feed_dict={state_input: x})
+
+    @property
+    def vectorized(self):
+        """Vectorized or not."""
+        return True
+
+    def dist_info_sym(self,
+                      obs_var,
+                      state_info_vars=None,
+                      name='dist_info_sym'):
+        """Symbolic graph of the distribution."""
+        with tf.variable_scope(
```

Comment:
i think this line should be removed and the scopes should be handled by `self.model.build()`